### PR TITLE
DCOS-40469: add status code to next event

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
BREAKING CHANGE: the output format of the next event changed. Instead of
`request(...).subscribe({next(response) {...}})` you can now write
`request(...).subscribe({next({ code, response }) {...}})`

### CODE REVIEW

File: src/__tests__/request-test.js [hunk 1]
```diff
@@ -71,12 +71,18 @@ describe("request", () => {
...
 
-      const connectionEventMock = { target: connectionMock };
+      const connectionEventMock = {
+        target: connectionMock
+      };
       connectionMock.__emit(ConnectionEvent.COMPLETE, connectionEventMock);
 
-      expect(observer.next).toHaveBeenCalledWith("some text");
+      expect(observer.next).toHaveBeenCalledWith({
+        response: "some text",
```
Comment (Poltergeist) [inline src/__tests__/request-test.js]: Can't we reference `connectionMock.response` here? and `connectionMock.xhr.status` in the next line? Not part of this PR a general question.

**Comment by nLight** [comment]
As discussed offline, let's please add `message` as well so that the interface is consistent across the abord

**Comment by DanielMSchmidt** [comment]
@nLight @Poltergeist fixed 👍
